lib/ansible/modules/crypto/luks_device.py

```diff
@@ -474,12 +467,15 @@ def run_module():
 
     # luks create
     if conditions.luks_create():
-        try:
-            crypt.run_luks_create(module.params['device'],
-                                  module.params['keyfile'])
-        except ValueError as e:
-            module.fail_json(msg="luks_device error: %s" % e)
+        if not module.check_mode:
+            try:
+                crypt.run_luks_create(module.params['device'],
+                                      module.params['keyfile'])
+            except ValueError as e:
+                module.fail_json(msg="luks_device error: %s" % e)
         result['changed'] = True
```

Comment:
```suggestion
        if not exists:
            result['changed'] = True
        else:
            result['changed'] = False
```

Reply:
IMO it might look like this

Reply:
This code is only executed if `module.params['device']` is not a LUKS container, so `result['changed']` has to be set to `True`.